cmd/operator-sdk/new/cmd.go

```diff
@@ -383,8 +388,29 @@ func verifyFlags() error {
 		if len(apiVersion) != 0 || len(kind) != 0 {
 			return fmt.Errorf("operators of type Go do not use --api-version or --kind")
 		}
-		if !makeVendor && projutil.DepManagerType(depManager) == projutil.DepManagerDep {
-			log.Warnf("--dep-manager=dep requires a vendor directory; ignoring --vendor=false")
+		inGopathSrc, err := projutil.WdInGoPathSrc()
+		if err != nil {
+			return err
+		}
+		switch projutil.DepManagerType(depManager) {
+		case projutil.DepManagerDep:
+			if !makeVendor {
+				log.Warnf("--dep-manager=dep requires a vendor directory; ignoring --vendor=false")
+			}
+			// dep assumes the project's path under $GOPATH/src is the project's
+			// repo path.
+			if repo != "" {
+				return fmt.Errorf(`--repo flag cannot be used with --dep-manger=dep`)
+			}
+			if !inGopathSrc {
+				return fmt.Errorf(`dependency manager "dep" requires the working directory to be in $GOPATH/src`)
+			}
+		case projutil.DepManagerGoMod:
+			if !inGopathSrc && repo == "" {
+				return fmt.Errorf(`dependency manager "modules" requires flag --repo be set if the working directory is not in $GOPATH/src`)
+			}
+		default:
+			return projutil.ErrInvalidDepManager(depManager)
```

Comment:
This set of changes looks almost identical to the changes in `migrate/cmd.go`. WDYT of leaving the `--vendor` flag check as is (so that the functions would be identical), and then exporting a function from `projutil` with this logic so that it can be shared by `new` and `migrate`?